firebase-firestore-ktx/src/main/kotlin/com/google/firebase/firestore/ktx/Firestore.kt

```diff
@@ -34,10 +34,24 @@ val Firebase.firestore
 /** Returns the [FirebaseFirestore] instance of a given [FirebaseApp]. */
 fun Firebase.firestore(app: FirebaseApp): FirebaseFirestore = FirebaseFirestore.getInstance(app)
 
-/** Returns the contents of the document converted to a POJO or null if the document doesn't exist. */
+/**
+ * Returns the contents of the document converted to a POJO or null if the document doesn't exist.
+ *
+ * @param T The Java class to create.
```

Comment:
Though this text exactly matches the corresponding Javadoc on `DocumentSnapshot` this doesn't seem right.

In the Javadoc, this is `@param valueType`, where it's describing that you actually have to pass the instance of `java.lang.Class` to make this work. For example, to get a value of type `Model`, you need to pass `Model.class`. The Javadoc is documenting that instance you pass, not the type parameter.

In the KDoc, the class instance is hidden behind a reified generic, and what you're documenting here is the type. Secondarily, we're going out of our way to hide the "java class" that you get via `T::class.java`.

A proposed alternative would be:

```
@param T The type of the object to create.
```

Here and throughout.

Reply:
Done